src/main/java/ca/warp7/frc2023/subsystems/Elevator.java

```diff
@@ -0,0 +1,121 @@
+package ca.warp7.frc2023.subsystems;
+
+import static ca.warp7.frc2023.Constants.*;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxAlternateEncoder;
+import com.revrobotics.SparkMaxLimitSwitch;
+import com.revrobotics.SparkMaxLimitSwitch.Type;
+import com.revrobotics.SparkMaxPIDController;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Elevator extends SubsystemBase {
+    private static Elevator instance;
+
+    // returns the instance of the subsystem
+    public static Elevator getInstance() {
+        if (instance == null) instance = new Elevator();
+        return instance;
+    }
+
+    public static CANSparkMax createMasterSparkMAX(int deviceID) {
+        CANSparkMax master = new CANSparkMax(deviceID, CANSparkMaxLowLevel.MotorType.kBrushed);
+        master.restoreFactoryDefaults();
+        master.setIdleMode(CANSparkMax.IdleMode.kBrake);
+        master.enableVoltageCompensation(12.0);
+        return master;
+    }
+
+    public CANSparkMax createFollowerSparkMAX(int deviceID) {
+        CANSparkMax follower = new CANSparkMax(deviceID, CANSparkMaxLowLevel.MotorType.kBrushed);
+        follower.restoreFactoryDefaults();
+        follower.follow(LeftMain);
+        follower.setIdleMode(CANSparkMax.IdleMode.kBrake);
+        follower.enableVoltageCompensation(12.0);
+        return follower;
+    }
+
+    // creates an inverted SparkMAx follower motor
+    public CANSparkMax createFollowerSparkMAXInverted(int deviceID) {
+        CANSparkMax follower = new CANSparkMax(deviceID, CANSparkMaxLowLevel.MotorType.kBrushed);
+        follower.restoreFactoryDefaults();
+        follower.follow(LeftMain, true);
+        follower.setIdleMode(CANSparkMax.IdleMode.kBrake);
+        follower.enableVoltageCompensation(12.0);
+        return follower;
+    }
+
+    // makes a leader and a follower motor for the left side
+    public CANSparkMax LeftMain = createMasterSparkMAX(kElevatorLeftMain);
+    public CANSparkMax LeftSecondary = createFollowerSparkMAX(kElevatorLeftSecondary);
+
+    // makes the inverted follower motors for the right side
+    public CANSparkMax RightMain = createFollowerSparkMAXInverted(kElevatorRightMain);
```

Comment:
There's not a main on both sides. My recommendation is to rename to leftFirst, leftSecond... Or something equivalent.

Reply:
@Skii1 if your are not able to do this then explain why